hydroflow_plus/src/location/cluster.rs

```diff
@@ -0,0 +1,227 @@
+use std::fmt::{Debug, Display};
+use std::hash::Hash;
+use std::marker::PhantomData;
+
+use proc_macro2::{Span, TokenStream};
+use quote::quote;
+use serde::{Deserialize, Serialize};
+use stageleft::runtime_support::FreeVariable;
+use stageleft::{quote_type, Quoted};
+
+// TODO(shadaj): have to user super due to stageleft limitations
```

Comment:
*use

Reply:
Also `super::super` 😭

Reply:
Yeah, need to fix `crate::` imports soon in general, big TODO for `stageleft`.